Prox/Prox/Models/Place.swift

```diff
@@ -451,6 +451,37 @@ struct OpenHours {
         return nil
     }
 
+    func getEventTimeText(forTime baseTime: Date) -> (String, String)? {
+        // Note: this function is basically a modified version of nextOpeningTime()
+        let midnight = DateComponents(hour: 0, minute: 0)
+
+        // HACK: Keeping this as a range so we can easily change this function look ahead more days if needed.
+        // (1...N) will look ahead N days after today.
+        let times = [baseTime] + (1...1).map { getDate(forTime: midnight, onDate: baseTime.addingTimeInterval(TimeInterval($0) * AppConstants.ONE_DAY))! }
+
+        for time in times {
+            guard let allOpeningTimes = getOpeningTimes(forDate: time),
+                  let openingPeriod = getCurrentOpeningPeriod(fromOpeningPeriods: allOpeningTimes, forDate: time) ?? getNextOpeningPeriod(fromOpeningPeriods: allOpeningTimes, forDate: time) else {
+                continue
+            }
+
+            let openingDayText: String
+            if OpenHours.calendar.isDate(time, inSameDayAs: baseTime) {
```

Comment:
I think this should be part of a refactor, returning a date and/or time object, and then allowing the consumer (PlaceDetails) to format the result however it needs. This would certainly fix the problem both here and in nextOpeningTime.